homeassistant/components/modbus/modbus.py

```diff
@@ -39,6 +49,14 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@lru_cache(maxsize=32)
+def _read_cached(client, what, unit, address, count, ttl_bucket):
```

Comment:
I really do not see the need for this added complexity, it does not seem to make your bit implementation easier, and it complicates the rest. I suggest to remove the lru_cache part (unless you have some good arguments, why not).

Reply:
If I have a configuration where each sensor/switch has the same address (in the case below it's 514) I have N read calls of the same Modbus register to the Modbus Client. If I have 16 sensors attached to the same address, the integration will end up with 16 reads of the same value through the Modbus.
This read cache holds the first read value for one second. This helps avoid making sequential calls. We can remove the cache, however, then we will be making the same read calls multiple times. This is not a big deal, however, may slow down updates.
```
    bit_sensors:
      - name: pwrmain_main_feed_under_over_voltage
        slave: 16
        address: 514
        bit_number: 1
      - name: pwrmain_controller_fault_watchdog_took_over
        slave: 16
        address: 514
        bit_number: 2
      - name: pwrmain_main_feed_phase_order_error
        slave: 16
        address: 514
        bit_number: 6
```

Reply:
Good idea. Let me think about how to do that. I will revert `_read_cached` changes